### PR TITLE
Add feature flag for enabling HTS auto creations

### PR DESCRIPTION
**Pull request by Neeharika-Sompalli**
Signed-off-by: Neeharika-Sompalli <neeharika.sompalli@swirldslabs.com>

**Description**:
Added a feature flag `tokens.autoCreations.isEnabled` to enable HTS  auto creations

Related to #3763

### CODE REVIEW

**Review by mhess-swl — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/grpc/marshalling/RoyaltyFeeAssessor.java [hunk 1]
```diff
@@ -74,7 +76,8 @@ public ResponseCodeEnum assessAllRoyalties(
                     // A NFT transfer with royalty fees to an unknown alias is not possible, since
                     // the auto-created
                     // account will not have any hbar to pay the fallback fee
-                    if (change.hasNonEmptyCounterPartyAlias()) {
+                    if (dynamicProperties.areTokenAutoCreationsEnabled()
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/grpc/marshalling/RoyaltyFeeAssessor.java:79]: Here also I think we can skip this check, and only do it once during `handleTransaction` in `AutoCreationLogic` as discussed.

Comment (Neeharika-Sompalli) [reply hedera-node/src/main/java/com/hedera/services/grpc/marshalling/RoyaltyFeeAssessor.java]: yeah agreed, changed to `AutoCreationLogic`. Thanks @tinker-michaelj

File: hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java [hunk 2]
```diff
@@ -180,6 +186,10 @@ public Pair<ResponseCodeEnum, Long> create(
             return Pair.of(MAX_ENTITIES_IN_PRICE_REGIME_HAVE_BEEN_CREATED, 0L);
         }
 
+        if (change.isForToken() && !properties.areTokenAutoCreationsEnabled()) {
```
Comment (tinker-michaelj) [inline hedera-node/src/main/java/com/hedera/services/txns/crypto/AutoCreationLogic.java:189]: So elegant! 😇

**Review by tinker-michaelj — APPROVED** [review]
LGTM!

**Pull request merged**